Lib/test/test_platform.py

```diff
@@ -741,5 +746,78 @@ def test_parse_os_release(self):
         self.assertEqual(len(info["SPECIALS"]), 5)
 
 
+class CommandLineTest(unittest.TestCase):
+    def setUp(self):
+        self.clear_caches()
+        self.addCleanup(self.clear_caches)
+
+    def clear_caches(self):
+        platform._platform_cache.clear()
+        platform._sys_version_cache.clear()
+        platform._uname_cache = None
+        platform._os_release_cache = None
+
+    @staticmethod
+    def text_normalize(string):
+        """Dedent *string* and strip it from its surrounding whitespaces.
+        This method is used by the other utility functions so that any
+        string to write or to match against can be freely indented.
+        """
+        return dedent(string).strip()
+
+    def invoke_platform(self, *flags):
+        output = io.StringIO()
+        with contextlib.redirect_stdout(output):
+            platform._main(args=flags)
+        return self.text_normalize(output.getvalue())
+
+    def test_unknown_flag(self):
+        with self.assertRaises(SystemExit):
+            # suppress argparse error message
+            with contextlib.redirect_stderr(io.StringIO()):
+                _ = self.invoke_platform('--unknown')
+
+    def test_invocation(self):
+        flags = (
+            "--terse", "--nonaliased", "terse", "nonaliased"
+        )
+
+        for r in range(len(flags) + 1):
+            for combination in itertools.combinations(flags, r):
+                self.invoke_platform(*combination)
+
+    def test_arg_parsing(self):
+        # Due to backwards compatibility, the `aliased` and `terse` parameters
+        # are computed based on a combination of positional arguments and flags.
+        #
+        # This test tests that the arguments are correctly passed to the underlying
+        # `platform.platform()` call. The parameters are two booleans for `aliased`
+        # and `terse`
+        options = (
+            ("--nonaliased", (False, False)),
+            ("nonaliased", (False, False)),
+            ("--terse", (True, True)),
+            ("terse", (True, True)),
+            ("nonaliased terse", (False, True)),
+            ("--nonaliased terse", (False, True)),
+            ("--terse nonaliased", (False, True)),
+        )
+
+        for flags, args in options:
+            with self.subTest(f"{flags}, {args}"):
+                with mock.patch.object(platform, 'platform') as obj:
+                    self.invoke_platform(*shlex.split(flags))
```

Comment:
Ideally, I would have done this manually in the test cases instead. Namely, `flags` should be `["--terse", "nonaliased"]` for instance.